dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -188,8 +191,8 @@ public async Task<SKContext> RunAsync(ContextVariables variables, CancellationTo
 
             try
             {
-                context.CancellationToken.ThrowIfCancellationRequested();
-                context = await f.InvokeAsync(context).ConfigureAwait(false);
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
This line is fine, but also shouldn't be necessary.

Reply:
It's just a way of bailing out of the function loop if cancellation is requested, instead of putting the onus of checking on the downstream function calls. Do you think we should let the functions themselves handle this instead? Or will the runtime bail out of the Task for us upon cancellation requested?

Reply:
It's fine to do this if we think it's likely the functions themselves won't be checking the token.  You could also do it in SKFunction.InvokeAsync to avoid doing this check anywhere InvokeAsync is called.